src/Mvc/Mvc.ViewFeatures/src/FormContext.cs

```diff
@@ -143,4 +158,20 @@ public void RenderedField(string fieldName, bool value)
 
         RenderedFields[fieldName] = value;
     }
+
+    internal bool InvariantField(string fieldName)
+    {
+        ArgumentNullException.ThrowIfNull(fieldName);
+
+        InvariantFields.TryGetValue(fieldName, out var result);
+
+        return result;
+    }
+
+    internal void InvariantField(string fieldName, bool value)
```

Comment:
nit; It is internal but maybe you could call it `MarkAsInvariantField` or something similar to let clear what it is doing.

Reply:
I did it this way to be consistent with `RenderedField` in this same class. Again, happy to change if we're ok with using slightly inconsistent naming.

Reply:
Oh! I should have checked the entire class before commenting 😂. Still unclear for me but I have never looked at this deeply before, so, I don't have a ton of context.